lib/_stream_readable.js

```diff
@@ -538,9 +538,9 @@ Readable.prototype.pipe = function(dest, pipeOpts) {
       // If the user unpiped during `dest.write()`, it is possible
       // to get stuck in a permanently paused state if that write
       // also returned false.
-      if (state.pipesCount === 1 &&
-          state.pipes[0] === dest &&
-          src.listenerCount('data') === 1 &&
+      // => Check whether `dest` is still a piping destination.
+      if (((state.pipesCount === 1 && state.pipes === dest) ||
+           (state.pipesCount > 1 && state.pipes.includes(dest))) &&
```

Comment:
It’s from ES2016, but it does what you think it does, i.e. `[1, 2, 3].includes(2) === true`. I like the clarity of it, but if you’d rather stick with `.indexOf(…)`, I’d definitely understand that.